pkgs/development/mobile/androidenv/emulate-app.nix

```
@@ -80,6 +82,14 @@ stdenv.mkDerivation {
           echo "hw.gpu.enabled=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
         ''}
 
+        echo "hw.lcd.density=${toString dpi}" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
+
+        ${lib.optionalString enableKeyboard ''
+          # Enable keyboard
+          echo "hw.keyboard=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
+          echo "hw.mainKeys=yes" >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
```

```suggestion
          {
             echo "hw.keyboard=yes"
             echo "hw.mainKeys=yes"
          } >> $ANDROID_SDK_HOME/.android/avd/device.avd/config.ini
```
Remove redundant comment, group echos to make it clearer that they're appended to the same file.